full-service/src/service/transaction.rs

```diff
@@ -351,7 +389,22 @@ where
                 builder.select_txos(&conn, max_spendable)?;
             }
 
-            let tx_proposal: TxProposal = builder.build(&conn)?;
+            let memo_builder: Box<dyn MemoBuilder + Send + Sync> = match memo {
```

Comment:
This should be a method in `TransactionMemo`, is that doable?

Reply:
hmm, yeah I don't see why not. I'll tinker around with that.

Reply:
Rust is cool. Thanks for prompting me on that, I keep forgetting I'm not writing in Swift.